www/js/splash/notifScheduler.ts

```diff
@@ -0,0 +1,269 @@
+import angular from 'angular';
+import React, { useEffect, useState } from 'react';
+import useAppConfig from '../useAppConfig';
+import { addStatReading, statKeys } from '../plugin/clientStats';
+import { getUser, updateUser } from '../commHelper';
+import { displayErrorMsg, logDebug } from '../plugin/logger';
+import { DateTime } from 'luxon';
+import i18next from 'i18next';
+
+let scheduledPromise = new Promise<void>((rs) => rs());
+let isScheduling = false;
+
+// like python range()
+function range(start, stop, step) {
+  let a = [start],
+    b = start;
+  while (b < stop) a.push((b += step || 1));
+  return a;
+}
+
+// returns an array of moment objects, for all times that notifications should be sent
+const calcNotifTimes = (scheme, dayZeroDate, timeOfDay) => {
+  const notifTimes = [];
```

Comment:
Type this as an array of `DateTime`s. Then TypeScript will be able to infer that as the return type of this function.